webui/react/src/components/TaskFilter.tsx

```diff
@@ -2,146 +2,113 @@ import { Select } from 'antd';
 import { SelectValue } from 'antd/es/select';
 import React, { useCallback, useMemo } from 'react';
 
-import Icon from 'components/Icon';
 import { CommandState, CommandType, RecentTask, RunState, TaskType, User } from 'types';
 import { capitalize } from 'utils/string';
 import { isExperimentTask } from 'utils/task';
-import { commandStateToLabel, runStateToLabel } from 'utils/types';
 
 import IconFilterButtons from './IconFilterButtons';
+import SelectFilter, { ALL_VALUE } from './SelectFilter';
+import StateSelectFilter from './StateSelectFilter';
 import css from './TaskFilter.module.scss';
+import UserSelectFilter from './UserSelectFilter';
 
-const { Option, OptGroup } = Select;
+const { Option } = Select;
 
-export const ALL_VALUE = 'all';
+export { ALL_VALUE };
 
-export interface TaskFilters {
+export interface TaskFilters<T extends TaskType = TaskType> {
   limit: number;
   states: string[];
   username?: string;
-  types: Record<TaskType, boolean>;
+  types: Record<T, boolean>;
 }
 
 interface Props {
   filters: TaskFilters;
   onChange: (filters: TaskFilters) => void;
-  authUser?: User;
-  users: User[];
+  showExperiments?: boolean;
```

Comment:
question: if we add comments here would they show up in IDE's IntelliSense?

Reply:
I gave this a shot by adding a comment that followed the property and see if it would show up on the consuming end and it didn't work for my via VSCode. :(